src/js/game/LevelMVC/LevelPlane.js

```diff
@@ -105,7 +105,7 @@ module.exports = class LevelPlane {
     }
 
     let redstoneToRefresh = [];
-    if (block.isRedstone || block.blockType === '' || block.isConnectedToRedstone) {
+    if (block.isRedstone || block.blockType === '' || (block.isConnectedToRedstone && !block.blockType.startsWith("piston"))) {
```

Comment:
this conditional is becoming a bit unwieldy, and it's not clear just by looking at it why pistons don't count. I vote we move this whole conditional into a method on LevelBlock that's well-named and -commented.

Reply:
👍